applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_ramm_arc_length_strategy.hpp

```diff
@@ -534,6 +541,23 @@ class GeoMechanicsRammArcLengthStrategy
         // Save the applied Lambda factor
         mLambda_old = mLambda;
     }
+
+private:
+    static double CalculateReferenceDofsNorm(DofsArrayType& rDofSet)
+    {
+        auto is_free_dof = [](const auto& rDof) { return rDof.IsFree(); };
+        auto free_dofs   = rDofSet | boost::adaptors::filtered(is_free_dof);
+
+        auto free_dof_values = Vector{boost::size(free_dofs)};
+        auto get_dof_value   = [](const auto& rDof) { return rDof.GetSolutionStepValue(); };
+        std::transform(std::begin(free_dofs), std::end(free_dofs), free_dof_values.begin(), get_dof_value);
+
+        return norm_2(free_dof_values);
+    }
```

Comment:
At some point, we could move this to a utility (it doesn't seem to be very arc length specific, but I might be wrong)

Reply:
You are correct that this calculation of the L2 norm of the (free) degrees of freedom is not specific for arc-length control. So it could be extracted to a more generic utility function and then be called from here. However, as we discussed, if we'd like to do that, let's do it in a separate PR.